app/controllers/WhatDoYouWantToDoController.scala

```diff
@@ -92,29 +92,47 @@ class WhatDoYouWantToDoController @Inject()(
     }
   }
 
+  private def showTaxCodeChangeTile(hasTaxCodeChanged: HasTaxCodeChanged): Boolean =
+    (hasTaxCodeChanged.changed, hasTaxCodeChanged.mismatch) match {
+      case (_, Some(mismatch)) if mismatch.confirmedTaxCodes.isEmpty => false
+      case (true, Some(TaxCodeMismatch(false, _, _)))                => true
+      case _                                                         => false
+    }
+
+  private def mostRecentTaxCodeChangeDate(nino: Nino, hasTaxCodeChanged: HasTaxCodeChanged)(
```

Comment:
nitpick: the name of this isnt to clear for me, as this does the showTaxCodeChangeTile check as well I think maybe displayTileWithDate or something like this would make more sense. I think this would read better in the view particularly.

Reply:
suggest a name and i'll change (although I think the name that should be changed is showTaxCodeChangeTile as this method just computes the mostRecentTaxCodeChangeDate)

Reply:
I like showTileWithDate, feel free to ignore :)

Reply:
changed